python_modules/dagster/dagster/_config/pythonic_config/resource.py

```diff
@@ -399,6 +399,7 @@ def _is_cm_resource_cls(cls: Type["ConfigurableResourceFactory"]) -> bool:
         return (
             cls.yield_for_execution != ConfigurableResourceFactory.yield_for_execution
             or cls.teardown_after_execution != ConfigurableResourceFactory.teardown_after_execution
+            or len(_get_resource_param_fields(cls)) > 0
```

Comment:
does this equate to "if there are any resources as params, assume they are context managers"? Is that the case because they always are in the base class?

Reply:
It basically equates to "if this resource has any other nested resources, let's assume that they're context managers, and treat this resource as a context manager"

This boolean affects a few things:
- The underlying resource function for the `ResourceDefinition` created from this `ConfigurableResource` - it's a CM function if `True`, not a CM if `False` (the source of the bug this PR addresses). This is largely opaque to the user, so biasing towards `True` here doesn't affect much.

- A few utility methods, like [`from_resource_context`](https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_config/pythonic_config/resource.py#L599) - this method is typically used for migrations from function-style resources which construct the Pythonic-style resource. These for the most part shouldn't use nesting:
  ```python
  class FancyDbResource(ConfigurableResource):
      conn_string: str
  
  @resource(config_schema=FancyDbResource.to_config_schema())
  def fancy_db_resource(context: InitResourceContext) -> FancyDbResource:
      return FancyDbResource.from_resource_context(context)
  ```

Reply:
got it - think its worth a comment